File: src/main/java/editor/gui/ccp/handler/EditorFrameTransferHandler.java
package editor.gui.ccp.handler;

import editor.gui.editor.EditorFrame;

/**
 * Transfer handler for importing cards into decks via the editor frame
 * (rather than a table in the frame).  Mainly used for convenience for
 * drag-and-drop.
 * 
 * @author Alec Roelke
 */
public class EditorFrameTransferHandler extends EditorTransferHandler
{
    /**
     * Create a new editor frame transfer handler.
     * 
     * @param e editor frame to handle transfers for
     * @param i ID of the list in the frame to handle transfers for
     */
    public EditorFrameTransferHandler(EditorFrame e, int i)
    {
        super(new EntryImportHandler(e, i), new CardImportHandler(e, i));
    }
}

File: src/main/scala/editor/gui/ccp/handler/EditorFrameTransferHandler.scala
package editor.gui.ccp.handler

import editor.gui.editor.EditorFrame

/**
 * Transfer handler for importing cards into decks via the [[EditorFrame]] itself, rather than the tables it contains. Mainly a convenience
 * for drag-and-drop.
 * 
 * @constructor create a new editor frame transfer handler for a particular list in a particular [[EditorFrame]]
 * @param editor frame to which the handler belongs
 * @param id ID of the list
 * 
 * @author Alec Roelke
 */
class EditorFrameTransferHandler(editor: EditorFrame, id: Int) extends EditorTransferHandler(Seq(EntryImportHandler(editor, id), CardImportHandler(editor, id)))